bin/packages/build.js

```diff
@@ -64,6 +64,36 @@ function createStyleEntryTransform() {
 	} );
 }
 
+function createBlockJsonEntryTransform() {
```

Comment:
I realize we might be aiming for the most direct solution, but:

- There's a huge amount of code duplication between this and the above function, which could be generalized.
- We should document every function; in this case, explaining the purpose of the transform.

Reply:
I've added some documentation, cheers for point that out. 

I'll push up a separate PR for exploring generalization of these transforms. I had an initial attempt, but I feel like it could go through some iteration and I wouldn't want to hold up this main PR.

Reply:
Here's the separate PR: #16317